scvi/model/_scvi.py

```diff
@@ -288,7 +288,4 @@ def to_latent_mode(
             )
         else:
             raise ValueError(f"Unknown latent mode: {mode}")
-        self.adata = reduced_adata
-        # This validates and sets a new adata manager
-        self.adata_manager.register_new_fields(self._get_latent_fields(mode))
-        self.module.latent_data_type = mode
+        self._update_adata_and_manager(reduced_adata, mode)
```

Comment:
could you add this change for scanvi also? (we should probably refactor this somehow..)